waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/metrics/MonitoringConfiguration.java

```diff
@@ -15,102 +15,94 @@
  */
 package com.hotels.bdp.waggledance.metrics;
 
-import java.io.Closeable;
-import java.net.InetSocketAddress;
-import java.util.HashSet;
-import java.util.Set;
-import java.util.concurrent.TimeUnit;
-
-import javax.annotation.PostConstruct;
-import javax.annotation.PreDestroy;
-
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
+import java.time.Duration;
+
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
-import com.codahale.metrics.JmxReporter;
-import com.codahale.metrics.MetricFilter;
-import com.codahale.metrics.MetricRegistry;
-import com.codahale.metrics.ScheduledReporter;
-import com.codahale.metrics.graphite.Graphite;
-import com.codahale.metrics.graphite.GraphiteReporter;
-import com.codahale.metrics.jvm.GarbageCollectorMetricSet;
-import com.codahale.metrics.jvm.MemoryUsageGaugeSet;
-import com.codahale.metrics.jvm.ThreadStatesGaugeSet;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.config.NamingConvention;
+import io.micrometer.core.instrument.util.HierarchicalNameMapper;
+import io.micrometer.graphite.GraphiteConfig;
+import io.micrometer.graphite.GraphiteMeterRegistry;
+import io.micrometer.graphite.GraphiteProtocol;
+import io.micrometer.jmx.JmxConfig;
+import io.micrometer.jmx.JmxMeterRegistry;
 
 import com.hotels.bdp.waggledance.conf.GraphiteConfiguration;
 
 @Configuration
 public class MonitoringConfiguration {
-  private static final Logger LOG = LoggerFactory.getLogger(MonitoringConfiguration.class);
-
-  private final Set<Closeable> reporters = new HashSet<>();
-  private @Autowired MetricRegistry metricRegistry;
-  private @Autowired GraphiteConfiguration graphiteConfiguration;
 
-  private <R extends Closeable> R registerReporter(R reporter) {
-    reporters.add(reporter);
-    return reporter;
-  }
+  private final GraphiteConfig DISABLED_GRAPHITE_CONFIG = new GraphiteConfig() {
+    @Override
+    public boolean enabled() {
+      return false;
+    }
 
-  @PreDestroy
-  public void destroy() {
-    if (reporters != null) {
-      for (Closeable reporter : reporters) {
-        try {
-          if (reporter instanceof ScheduledReporter) {
-            ((ScheduledReporter) reporter).report();
-          }
-          reporter.close();
-        } catch (Exception e) {
-          LOG.warn("Problem closing reporter", e);
-        }
-      }
+    @Override
+    public String get(String key) {
+      return null;
     }
-  }
+  };
 
-  @PostConstruct()
-  public void init() {
-    registerBaseMetrics();
-    registerReporter(jmxReporterBuilder().build()).start();
+  @Bean
+  public GraphiteMeterRegistry graphteMeterRegistry(GraphiteConfiguration graphiteConfiguration) {
```

Comment:
Doesn't this mean that users now can't provide a meter registry that uses anything other than Graphite as this is now bound to the concrete type? This is probably OK for now as we've only offered turning it on/off which this seems to achieve but I guess if we move to full Spring Boot Actuator for this we'll remove the beans and config from here entirely?

Reply:
I need to read up more on the actuator to answer your last question but on the first, no
This is how the docs say to override the GraphiteMeterRegistry.
You inject MeterRegistry in the classes where you want to use them and it will be a Composite containing all defined MeterRegistries for our case ( a Graphite and JMX one). So if you want ganglia(?) you add the corresponding jar to your classpath and spring boot hooks it all up and gives you that as well.
Because we add the graphite jar in our classpath we'll always have graphite in hence we need to disable it.